internal/generator/vector/output/common/auth/bearer_token.go

```diff
@@ -23,11 +25,28 @@ token = "{{.Token}}"
 `
 }
 
-func NewBearerToken(id string, spec *obs.HTTPAuthentication, secrets helpers.Secrets) BearerToken {
+func NewBearerToken(id string, auth *obs.HTTPAuthentication, secrets helpers.Secrets) BearerToken {
 	bt := BearerToken{}
-	if spec != nil {
+	if auth != nil && auth.Token != nil && auth.Token.Secret != nil {
 		bt.ID = id
-		bt.Token = secrets.AsStringFromBearerToken(spec.Token)
+		key := &obs.SecretKey{
+			Secret: &corev1.LocalObjectReference{
+				Name: auth.Token.Secret.Name,
+			},
+			Key: auth.Token.Secret.Key,
+		}
+		bt.Token = helpers.SecretFrom(key)
+		return bt
 	}
+	//TODO fixme
+
+	// We reconcile SA token secret, so name and key will be known
+	key := &obs.SecretKey{
```

Comment:
This is where once #2541 merges we should fetch token from the SECRET vault, but it will need to retrieve it from the projected SA token volume.  All the other values are in the secret but this one is sourced from the file system